.github/scripts/utils.py/check-services.py

```diff
@@ -265,25 +208,47 @@ def main():
             continue
 
         service_type = service.get("recommended", {}).get("output", "rtmp_output")
-        if service_type not in {"rtmp_output", "ffmpeg_hls_muxer", "ftl_output"}:
+        if service_type not in {"rtmp_output", "ffmpeg_hls_muxer"}:
             print("Unknown service type:", service_type)
             new_services["services"].append(service)
             continue
 
+        protocol = "rtmp" if service_type == "rtmp_output" else "hls"
+
         # create a copy to mess with
         new_service = service.copy()
         new_service["servers"] = []
 
         # run checks for all the servers, and store results in timestamp cache
-        for server in service["servers"]:
-            if service_type == "ftl_output":
-                is_ok = check_ftl_server(server["url"])
-            elif service_type == "ffmpeg_hls_muxer":
-                is_ok = check_hls_server(server["url"])
-            else:  # rtmp
-                is_ok = check_rtmp_server(server["url"])
+        try:
+            servers = [s["url"] for s in service["servers"]]
+            tasks = []
+            for host in check_servers:
+                tasks.append(
+                    asyncio.create_task(
+                        check_servers_task(session, host, protocol, servers)
+                    )
+                )
+            results = await asyncio.gather(*tasks)
+        except Exception as e:
+            print(
+                f"❌ Querying server status for \"{service['name']}\" failed with: {e}"
+            )
+            return 1
+
+        # go over results
+        for server, result in zip(service["servers"], zip(*results)):
+            failure_count = sum(not res["status"] for res in result)
+            probe_count = len(result)
+            # only treat server as failed if all check servers reported a failure
+            is_ok = failure_count < probe_count
 
             if not is_ok:
+                failures = {res["comment"] for res in result if res["comment"]}
+                print(
+                    f"⚠️ Connecting to server failed: {server['url']} (Reason(s): {failures})"
```

Comment:
Since the URL is considered a secret key by github, this message will be censored automatically. Is there an interface where privileged members can see the actual value? If not, you could perform a `server['url'].split('.', 1)[1]` to get the region name from the URL.

Reply:
The URL here is the streaming server URL, not the checker URL. If any of them fail that's currently treated as a failure of the script and it will exit.

Reply:
Updated to hide the prefix and suffix and leave the region code. Tested on my repo to confirm it works as expected.